examples/with-google-analytics/pages/_app.js

```diff
@@ -1,43 +1,28 @@
-import { useEffect } from 'react'
+import { useState } from 'react'
 import Script from 'next/script'
 import { useRouter } from 'next/router'
-import * as gtag from '../lib/gtag'
 
 const App = ({ Component, pageProps }) => {
+  const [ga, setGa] = useState(null);
   const router = useRouter()
-  useEffect(() => {
-    const handleRouteChange = (url) => {
-      gtag.pageview(url)
-    }
-    router.events.on('routeChangeComplete', handleRouteChange)
-    return () => {
-      router.events.off('routeChangeComplete', handleRouteChange)
-    }
-  }, [router.events])
+
+  function handleLoad() {
+    const {ga} = window;
+
+    setGa(() => ga)
```

Comment:
Could you explain why this way instead of what was here before?

Reply:
The idea is to guide users to store those global variables in places where they have control. Like the example about stripe in `next/script`, doing this with the `onLoad` prop from `next/script` seems to be the `safer` way to ensure that the global variable is ready to be stored